readthedocs/rtd_tests/tests/test_doc_building.py

```diff
@@ -1614,10 +1618,16 @@ def test_is_obsolete_with_json_same_data_as_version(self, load_config):
             build_env=self.build_env,
             config=config,
         )
-        env_json_data = '{"build": {"image": "readthedocs/build:2.0", "hash": "a1b2c3"}, "python": {"version": 3.5}, "envvars_hash": 1234}'  # noqa
-        with patch('os.path.exists') as exists, patch('readthedocs.doc_builder.python_environments.PythonEnvironment._get_envvars_hash') as get_hash, patch('readthedocs.doc_builder.python_environments.open', mock_open(read_data=env_json_data)) as _open:  # noqa
+
+        self.assertFalse(self.pip.environmentvariable_set.all().exists())
+        get(EnvironmentVariable, project=self.pip, name='ABCD', value='1234')
+        env_vars_hash = hash((
+            ('ABCD', '1234'),
+        ))
+
+        env_json_data = '{"build": {"image": "readthedocs/build:2.0", "hash": "a1b2c3"}, "python": {"version": 3.5}, "env_vars_hash": %s}' % env_vars_hash  # noqa
```

Comment:
nit: don't use module formatter here. Use f-string or `.format`.

Reply:
We cannot use `.format` or `f-string` because the string is in dict format and contains multiple `{` which produces an error when these formats are used.

Reply:
You can just double the `{` https://docs.python.org/3/library/string.html#formatstrings

Reply:
@stsewd 
Thanks for this.
I have updated the PR.